cterasdk/core/types.py

```diff
@@ -54,3 +58,117 @@ class GroupAccount(PortalAccount):
     @property
     def account_type(self):
         return PortalAccountType.Group
+
+
+class ShareRecipient:
```

Comment:
I like the notion of returning self in order to execute everything in one line. One thing I think should be different it that the account type methods should be implemented as factory methods and hence should be marked as staticmethod, create the object, set the fields are return it. 
This way the usage code will be: `ShareRecipient.exteral(test@exmaple.com).read_write()` 
instead of `ShareRecipient().exteral(test@exmaple.com).read_write()`

Reply:
Fixed.